examples/cpp/pyperf/PyPerfProfiler.cc

```diff
@@ -230,24 +234,26 @@ Populate the PidData BPF map.
 */
 bool PyPerfProfiler::populatePidTable() {
   bool result = false;
-
   // Populate config for each Python Process
   auto pid_config_map = bpf_.get_hash_table<int, PidData>(kPidCfgTableName);
-
   logInfo(3, "Pruning dead pids\n");
   auto pid_config_keys = pid_config_map.get_keys_offline();
   for (const auto pid : pid_config_keys) {
     auto pos = std::find(pids.begin(), pids.end(), pid);
     if (pos == pids.end()) {
+      // Remove dead pid from config map and native stack trace's cache
       pid_config_map.remove_value(pid);
+      NativeStackTrace::prune_dead_pid(pid);
     }
     else {
       result = true;
+      // To avoid re-population
```

Comment:
Don't we ever `update_value` of existing PIDs?

I looked at the fields:
```
typedef struct pid_data {
  enum pthreads_impl pthreads_impl;
  struct py_globals globals;
  struct struct_offsets offsets;
  uintptr_t interp;  // vaddr of PyInterpreterState
} PidData;
```

from a skim on `tryTargetPid`, it seems to work all-or-nothing, so if it returned `true` we know all fields are filled in, and we won't need re-update because those fields don't change for the same process. Did you consider it further? If not, we need to inspect if this logic change makes sense.

Btw what caused you to make the change, anyway? (it appears to be a bug in the logic but I wonder how you arrived at it :thinking: )

Reply:
Hmm, I'm not sure on which part I originally wrote it. I think I might have mis-read the diff and thought that you added the `pids.erase(pos)` call in this PR? I now see that you didn't, and the logic appears sound to me (1. get running python pids; 2. prune dead pids from kernel map; 2. prune pids already in kernel map from pids list; 4. adds pythons in pids list to kernel map) . You're welcome to add a bit more of documentation around here as you've added, on the flow of this function, since it's a complex one. This comment is resolved IMO, leaving it open for you to read.